### PR TITLE
BUG: #31464 Fix error when parsing JSON list of bool into Series

### LINKED ISSUE #31464: read_json with typ="series" of json list of bools results in timestamps/Exception
#### Code Sample, a copy-pastable example if possible

```python
import pandas as pd
pd.read_json('[true, true, false]', typ="series")
```
results in the following Pandas Series object in older Pandas versions:
```
0   1970-01-01 00:00:01
1   1970-01-01 00:00:01
2   1970-01-01 00:00:00
dtype: datetime64[ns]
```
Since 1.0.0 it raises TypeError: <class 'bool'> is not convertible to datetime

#### Problem description
The expected output would be a Pandas Series of bools. Note that
* with typ="frame" it works and the result is a dataframe with one column with bool values
* with convert_dates set to False correctly outputs a Series of boolean values

This is a problem because
* users would expect a Series of bools (and neither an exception nor a series of timestamps)
* it is inconsistent with the "frame" case

#### Output of ``pd.show_versions()``

<details>

[paste the output of ``pd.show_versions()`` here below this line]

INSTALLED VERSIONS
------------------
commit           : None
python           : 3.8.1.final.0
python-bits      : 64
OS               : Linux
OS-release       : 5.4.13-arch1-1
machine          : x86_64
processor        : 
byteorder        : little
LC_ALL           : None
LANG             : de_DE.UTF-8
LOCALE           : de_DE.UTF-8

pandas           : 1.0.0
numpy            : 1.18.1
pytz             : 2019.3
dateutil         : 2.8.1
pip              : 20.0.2
setuptools       : 44.0.0
Cython           : 0.29.14
pytest           : 5.2.4
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : 4.4.2
html5lib         : 1.0.1
pymysql          : None
psycopg2         : None
jinja2           : 2.10.3
IPython          : 7.11.1
pandas_datareader: None
bs4              : None
bottleneck       : None
fastparquet      : None
gcsfs            : None
lxml.etree       : 4.4.2
matplotlib       : 3.1.2
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pytables         : None
pytest           : 5.2.4
pyxlsb           : None
s3fs             : None
scipy            : 1.3.2
sqlalchemy       : 1.3.11
tables           : None
tabulate         : None
xarray           : None
xlrd             : 1.2.0
xlwt             : None
xlsxwriter       : None
numba            : None
</details>

### PR DESCRIPTION
**Pull request by jessefarnham**
Add a missing exception type to the except clause, to cover
the TypeError that is thrown by Cythonized array_to_datetime
function when trying to convert bool to nonseconds.

- [x] closes #31464
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: pandas/io/json/_json.py [hunk 1]
```diff
@@ -982,7 +982,7 @@ def _try_convert_to_date(self, data):
         for date_unit in date_units:
             try:
                 new_data = to_datetime(new_data, errors="raise", unit=date_unit)
-            except (ValueError, OverflowError):
+            except (ValueError, OverflowError, TypeError):
```
Comment (WillAyd) [inline pandas/io/json/_json.py:985]: @jbrockmendel any thoughts here?

**Review by WillAyd — COMMENTED** [review]
Thanks for the PR! I think looks pretty good

File: pandas/io/json/_json.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/io/json/_json.py:985]: the usual: if we can handle this case before the try/except that would be preferable

Comment (jessefarnham) [reply pandas/io/json/_json.py:985]: @jbrockmendel I’ll give that approach a try.

Comment (jessefarnham) [reply pandas/io/json/_json.py]: @jbrockmendel Pushed an alternate approach that escapes from the _try_convert_to_date() method if the dtype is bool, avoiding the try/except but adding a conditional to the method (which I don't love, but at least the method already has dtype-based conditionals).

File: pandas/io/json/_json.py [hunk 2]
```diff
@@ -978,6 +978,10 @@ def _try_convert_to_date(self, data):
             if not in_range.all():
                 return data, False
 
+        # ignore bool
+        if new_data.dtype == "bool":
+            return data, False
```
Comment (jbrockmendel) [inline pandas/io/json/_json.py:983]: can you add a comment inside the if `# GH#33373 ignore bool`

Comment (jessefarnham) [reply pandas/io/json/_json.py:983]: Done. Test failures appear to be unrelated; the FDIC site that the html tests use is returning intermittent 404s.

Comment (WillAyd) [reply pandas/io/json/_json.py:983]: Hmm actually not sure about doing it this way; we've done this type of checking in groupby and its not a great precedent. We do already catch the three errors a few lines above this - shouldn't we just do the same here?

Comment (jessefarnham) [reply pandas/io/json/_json.py:983]: The first approach seems a bit simpler to me—this seems to be a case where whoever put the TypeError into the other except clauses accidentally omitted it from the one that I originally added it to. Let me know what you and @jbrockmendel decide and I’ll be happy to update accordingly.

Comment (jbrockmendel) [reply pandas/io/json/_json.py:983]: im happy to defer to @WillAyd in this part of the code.  catching TypeError is definitely more idiomatic python, but we've definitely seen how catching too much has let bugs remain un-surfaced for a long time elsewhere

Comment (WillAyd) [reply pandas/io/json/_json.py:983]: Yea I think should catch the TypeError as done originally; would match what we have on L968 anyway

Comment (jessefarnham) [reply pandas/io/json/_json.py]: Sounds good -- pushed the revert back to the original change. Thanks for the quick turnaround -- let me know if anything else needed.

File: pandas/tests/io/json/test_readlines.py [hunk 3]
```diff
@@ -179,3 +179,9 @@ def test_readjson_unicode(monkeypatch):
         result = read_json(path)
         expected = pd.DataFrame({"£©µÀÆÖÞßéöÿ": ["АБВГДабвгд가"]})
         tm.assert_frame_equal(result, expected)
+
+
+def test_readjson_bool_series():
```
Comment (WillAyd) [inline pandas/tests/io/json/test_readlines.py:184]: Can you move this to `test_pandas` and add `GH 31464` as a comment in the function body?

**Review by WillAyd — CHANGES_REQUESTED** [review]
minor nit otherwise lgtm

File: pandas/tests/io/json/test_readlines.py (hunk 3 above)
Comment (jessefarnham) [reply pandas/tests/io/json/test_readlines.py]: Done

**Review by WillAyd — APPROVED** [review]
lgtm

File: pandas/io/json/_json.py (hunk 1 above)
Comment (jreback) [reply pandas/io/json/_json.py:985]: yes this is ok. an alternative approach is to `infer_dtype` on this and then dispatch. If someone wants to try that (later PR) would be ok.

**Comment by jreback** [comment]
thanks @jessefarnham very nice.

**Pull request merged**